Source/CarthageKit/FrameworkExtensions.swift

```diff
@@ -154,3 +155,72 @@ extension NSScanner {
 		return nsString.substringWithRange(lineRange)
 	}
 }
+
+extension NSURLSession {
+	/// Returns a signal that will download a file using the given request. The
+	/// file will be deleted after the URL has been sent upon the signal.
+	internal func rac_downloadWithRequest(request: NSURLRequest) -> ColdSignal<(NSURL, NSURLResponse)> {
+		return ColdSignal { (sink, disposable) in
+			let serialDisposable = SerialDisposable()
+			let handle = disposable.addDisposable(serialDisposable)
+
+			let task = self.downloadTaskWithRequest(request) { (URL, response, error) in
+				// Avoid invoking cancel(), or the download may be deleted.
+				handle.remove()
+
+				if URL == nil || response == nil {
+					sink.put(.Error(error))
+				} else {
+					let value = (URL!, response!)
+					sink.put(.Next(Box(value)))
+					sink.put(.Completed)
+				}
+			}
+
+			serialDisposable.innerDisposable = ActionDisposable {
+				task.cancel()
+			}
+
+			task.resume()
+		}
+	}
+}
+
+extension NSURL: JSONDecodable {
+	public class func decode(json: JSONValue) -> Self? {
+		if let URLString = String.decode(json) {
+			return self(string: URLString)
+		} else {
+			return nil
+		}
+	}
+}
+
+extension NSFileManager {
+	/// Creates a directory enumerator at the given URL. Sends each URL
+	/// enumerated, along with the enumerator itself (so it can be introspected
+	/// and modified as enumeration progresses).
+	internal func rac_enumeratorAtURL(URL: NSURL, includingPropertiesForKeys keys: [String], options: NSDirectoryEnumerationOptions, catchErrors: Bool = false) -> ColdSignal<(NSDirectoryEnumerator, NSURL)> {
```

Comment:
> If this always caught errors, then the signal would terminate, and resuming the enumeration from that point would be a pain, yes?

You mean if it passed through errors? That's correct.

> I really think this should be `carthage_…` instead of `rac_…`. I know you’re not likely to step on your own toes with this one, but it does sort of defeat the purpose of the prefix to use another project’s :smile:

I've been using these as a testbed for methods that could maybe get lifted into RAC.